tower_eval/cli.py

```diff
@@ -121,28 +121,31 @@ def run_evaluations(configs: dict) -> dict:
             task_name = task.get("name")
             subtasks = task.get("subtasks")
             task_metrics = task.get("metrics")
-            for subtask, subtask_args in subtasks.items():
-                subtask_results = {}
-                # Subtasks can overwrite/add some arguments to the metric argument defined on the task level
-                # For example, we should be able to define a specific tokenizer for Chinese only
-                if subtask_args is None:
-                    subtask_args = {}
-                subtask_metrics = (
-                    subtask_args.get("metrics", {}) if subtask_args is not None else {}
-                )
-                output_path = (
-                    output_dir
-                    / task_name
-                    / subtask
-                    / model["type"]
-                    / model["name"]
-                    / "evaluation.json"
-                )
-                dataset_name, language = subtask.split(".")
-                for task_metric, task_metric_args in task_metrics.items():
+            for task_metric, task_metric_args in task_metrics.items():
+                task_metric_args = {} if task_metric_args is None else task_metric_args
+                instantiated_metric = available_metrics[task_metric](**task_metric_args)
+                for subtask, subtask_args in subtasks.items():
                     logger.opt(colors=True).info(
                         f"Evaluating the results of model <green> {model_name} </green> on task: <yellow> {task_name} </yellow>, subtask: <green> {subtask} </green> with metric: <red> {task_metric} </red>"
                     )
+                    subtask_results = {}
+                    # Subtasks can overwrite/add some arguments to the metric argument defined on the task level
+                    # For example, we should be able to define a specific tokenizer for Chinese only
+                    if subtask_args is None:
```

Comment:
I think this is redundant since in line 139 we repeat the same thing.

Reply:
is it? in line 139 we define subtask_metrics, not args or results